File: List/旋转数组的最小数字.cpp
class Solution {
public:
    int minNumberInRotateArray(vector<int> rotateArray) {
        int left = 0;
        int right = rotateArray.size()-1;
        int resIndex;
        while(rotateArray[left] >= rotateArray[right]){
            if(right-left == 1){
                resIndex = right;
                break;
            }
            int mid = (left+right)/2;
            if(rotateArray[left] <= rotateArray[mid]){
                left = mid;
            }
            else if(rotateArray[right] >= rotateArray[mid]){
                right = mid;
            }
        }
        return rotateArray[resIndex];
    }
};

File: List/旋转数组的最小数字.py
# -*- coding:utf-8 -*-
class Solution:
    def minNumberInRotateArray(self, rotateArray):
        # write code here
        left = 0
        right = len(rotateArray)-1
        while rotateArray[left] >= rotateArray[right]:
            if right-left == 1:
                return rotateArray[right]
            mid = (right + left) / 2
            if rotateArray[mid] >= rotateArray[left]:
                left = mid
            elif rotateArray[mid] <= rotateArray[right]:
                right = mid